### PR TITLE
Publisher UI modifications.

### PR DESCRIPTION
**Pull request by menakaj**
This PR contains following modifications.

1. Created custom DataTable component.
2. Added application and platform create form validations.
3. Added image visualization feature for screenshots etc. 
4. Removed the overview section.

### CODE REVIEW

File: components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/src/components/Platform/PlatformListing.jsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    _compare(a, b) {
+        if (a.applicationName < b.applicationName)
+            return -1;
+        if (a.applicationName > b.applicationName)
+            return 1;
+        return 0;
+    }
+
+    _onRowClick(id) {
+        console.log(id)
```
Comment (charithag) [inline components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/src/components/Platform/PlatformListing.jsx:73]: Please remove unnecessary console logs

File: components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/src/components/UIComponents/DataTable.jsx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        }
+
+    };
+
+    componentWillMount() {
+        this.setState({data: this.props.data, headers: this.props.headers})
+    }
+
+    shouldComponentUpdate(nextProps, nextState) {
+        console.log("next Props", nextProps.data);
```
Comment (charithag) [inline components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/src/components/UIComponents/DataTable.jsx:67]: Please remove unnecessary console logs

File: components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/src/components/UIComponents/DataTable.jsx [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+     * This method invokes the parent method handleRowClick, which is passed via props.
+     * */
+    _handleRowClick(id) {
+        this.props.handleRowClick(id);
+    }
+
+    render() {
+        const {data, headers} = this.state;
+
+        console.log(data);
```
Comment (charithag) [inline components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/src/components/UIComponents/DataTable.jsx:83]: Please remove unnecessary console logs

**Review by charithag — COMMENTED** [review]
Please refactor the code along with comments

File: components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/src/components/Platform/PlatformListing.jsx (hunk 1 above)
Comment (menakaj) [reply components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/src/components/Platform/PlatformListing.jsx:73]: This is already fixed in later commits.
[https://github.com/wso2/carbon-device-mgt/pull/1081](url)

File: components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/src/components/UIComponents/DataTable.jsx (hunk 2 above)
Comment (menakaj) [reply components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/src/components/UIComponents/DataTable.jsx:67]: This is already fixed in commit, 
https://github.com/wso2/carbon-device-mgt/commit/adad9b307fac2faef7b35aaa4d4c0e699d8cc3e0

File: components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/src/components/UIComponents/DataTable.jsx (hunk 3 above)
Comment (menakaj) [reply components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/src/components/UIComponents/DataTable.jsx:83]: This is already fixed in, 
https://github.com/wso2/carbon-device-mgt/commit/adad9b307fac2faef7b35aaa4d4c0e699d8cc3e0

**Pull request merged**